app/src/main/cpp/skyline/services/audout/audout.cpp

```diff
@@ -0,0 +1,110 @@
+#include "audout.h"
+#include <kernel/types/KProcess.h>
+
+namespace skyline::service::audout {
+    audoutU::audoutU(const DeviceState &state, ServiceManager &manager) : BaseService(state, manager, false, Service::audout_u, {
+        {0x0, SFUNC(audoutU::ListAudioOuts)},
+        {0x1, SFUNC(audoutU::OpenAudioOut)}
+    }) {}
+
+    void audoutU::ListAudioOuts(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        state.thisProcess->WriteMemory((void*)audio::constant::DefaultAudioOutName, request.outputBuf.at(0).address, sizeof(audio::constant::DefaultAudioOutName));
+    }
+
+    void audoutU::OpenAudioOut(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        u32 sampleRate = request.Pop<u32>();
+        u16 channelCount = static_cast<u16>(request.Pop<u32>() >> 0xf);
+
+        state.logger->Info("audoutU: {}", __func__);
+        state.logger->Info("Sample rate: {}", sampleRate);
+        state.logger->Info("Channel count: {}", channelCount);
+
+        if (!sampleRate)
+            sampleRate = audio::constant::SampleRate;
+
+        if (!channelCount)
+            channelCount = audio::constant::ChannelCount;
```

Comment:
You can use the ternary operator to break both of these into a single statement:
`sampleRate = sampleRate : sampleRate ? audio::constant::SampleRate;`
`channelCount = channelCount : channelCount ? audio::constant::ChannelCount;`

Reply:
Android studio complains about assigning a variable to itself with this. You know of a way to silence the warning or should I leave as it?